c2rust-transpile/src/rust_ast/mod.rs

```diff
@@ -28,14 +27,6 @@ pub fn pos_to_span(pos: BytePos) -> Span {
     SpanExt::new(pos.0, pos.0)
 }
 
-pub const DUMMY_SP: Span = unsafe {
-    std::mem::transmute(SpanRepr {
-        compiler_or_fallback: 1,
-        lo: 0,
-        hi: 0,
-    })
-};
-
```

Comment:
If we remove `DUMMY_SP`, do we still need `SpanRepr` (I'm not sure)?

Reply:
Yes, unfortunately. It is still used in `get_inner` and `synthesize`. It is still UB. At the moment I have no solution.

Reply:
It seems `get_inner` could be implemented inefficiently and very unstably, but soundly, by parsing `Span`'s `Debug` representation.  And `get_inner_mut` is unused, but `synthesize` I'm not sure how to do without UB.  Also, `SpanRepr` should be the same source code as `Span`, which would minimize chances of it breaking in practice, as `rustc` is unlikely to compile the same source code with different ABIs.